src/ORM/Behavior/TranslateBehavior.php

```diff
@@ -272,6 +272,43 @@ public function beforeSave(Event $event, EntityInterface $entity, ArrayObject $o
         $newOptions = [$this->_translationTable->getAlias() => ['validate' => false]];
         $options['associated'] = $newOptions + $options['associated'];
 
+        // Check early if empty transaltions are present in the entity.
+        // If this is the case, unset them to prevent persistence.
+        // This only applies if $this->_config['allowEmptyTranslations'] is false
+        if ($this->_config['allowEmptyTranslations'] === false) {
+            $translations = (array)$entity->get('_translations');
+            if (!empty($translations)) {
```

Comment:
Foreaching over the empty set is fine and saves you a level of indentation/complexity.